libs/entitlement/bridge/src/main/java/org/elasticsearch/entitlement/bridge/EntitlementChecker.java

```diff
@@ -9,6 +9,20 @@
 
 package org.elasticsearch.entitlement.bridge;
 
+import java.net.URL;
+import java.net.URLStreamHandlerFactory;
+
 public interface EntitlementChecker {
     void check$java_lang_System$exit(Class<?> callerClass, int status);
+
+    // URLClassLoader ctor
+    void check$java_net_URLClassLoader$(Class<?> callerClass, URL[] urls);
+
+    void check$java_net_URLClassLoader$(Class<?> callerClass, URL[] urls, ClassLoader parent);
+
+    void check$java_net_URLClassLoader$(Class<?> callerClass, URL[] urls, ClassLoader parent, URLStreamHandlerFactory factory);
+
+    void check$java_net_URLClassLoader$(Class<?> callerClass, String name, URL[] urls, ClassLoader parent);
+
+    void check$java_net_URLClassLoader$(Class<?> callerClass, String name, URL[] urls, ClassLoader parent, URLStreamHandlerFactory factory);
```

Comment:
I wonder if there would be a way to check if they call other ctors and just use the one that is called since I would guess a lot of these end just passing along default arguments.

Another thought is we just assume all non-private ctors require this check instead of having to list out all of the args and construct what we need that way. Maybe that doesn't work out easily, though.

Reply:
> if there would be a way to check if they call other ctors and just use the one that is called since I would guess a lot of these end just passing along default arguments.

Good point! 
We can check, by hand :) or extend https://github.com/elastic/elasticsearch/pull/116257 to do that.
I'll check by hand for this case, and remove the ones that do just `this(...)`

Reply:
Actually, they all call `super()`: https://github.com/openjdk/jdk22/blob/fe9f05023e5a916b21e2db72fa5b1e8368a2c07d/src/java.base/share/classes/java/net/URLClassLoader.java
But this is definitely something to keep in mind when implementing checks on this kind of extra-overloaded constructors or methods!